express/api_routes.js

```diff
@@ -1,16 +1,20 @@
 'use strict';
-const { extension } = require('./extension');
 const express = require('express');
 const { sessionMiddleware } = require('./middleware/session_middleware');
 const { ApplicationConfig, ApplicationClient } = require("@gofynd/fdk-client-javascript");
+const { ExtensionFactory } = require('./extension_factory');
 
 
-function setupProxyRoutes() {
+function setupProxyRoutes(extension) {
     const apiRoutes = express.Router({  mergeParams: true });
     const applicationProxyRoutes = express.Router({  mergeParams: true });
 
-    applicationProxyRoutes.use(async (req, res, next) => {
+    applicationProxyRoutes.use(["/:cluster_id", "/"],async (req, res, next) => {
         try {
+            const clusterId = req.params.cluster_id;
+            if (clusterId) {
+                extension = ExtensionFactory.getExtension(clusterId)
+            }
```

Comment:
`req.extension = extension;`

Injection of _extension_ instance in the _request_ object will be helpful to get the _cluster_id_